### PR TITLE
Guido/fix from dict

### PR DESCRIPTION
**Pull request by gkoller**
A deepcopy is required if we use a `dict` as the basis of our datastructures. Afterall we don't want our internal representation to be altered because the outside world still had a reference to the `dict` that was originally passed in to `from_dict`

Extracted `SemVer` into `more_ds`

### CODE REVIEW

File: src/schematools/cli.py [hunk 1]
```diff
@@ -363,7 +364,9 @@ def validate(
     structural_errors = False
     try:
         click.echo("Structural validation: ", nl=False)
-        jsonschema.validate(instance=dataset.json_data(), schema=meta_schema)
+        jsonschema.validate(
```
Comment (jjmurre) [inline src/schematools/cli.py:367]: Nice addition!

**Review by jjmurre — APPROVED** [review]
LGTM. Nice that SemVer is now widely available.

**Comment by lbam** [comment]
I'd like to see a test for the deepcopy.

**Comment by gkoller** [comment]
> I'd like to see a test for the deepcopy.

Will add test.

**Comment by gkoller** [comment]
@lbam The test proofed valuable, as it turns out de deepcopy doesn't do anything valuable here. Reading up on `UserDict` (a base class of `DatasetSchema`)  I see that:

>  The instance’s contents are kept in a regular dictionary, which is accessible via the data attribute of UserDict instances. If initialdata is provided, data is initialized with its contents; note that a reference to initialdata will not be kept, allowing it to be used for other purposes.

So a deepcopy is unnecessary. I gues I have to look at another fix for the failing tests in `dso-api`.

**Comment by lbam** [comment]
Maybe push the test without the changes?

**Pull request merged**